src/UCTNode.cpp

```diff
@@ -103,8 +104,18 @@ bool UCTNode::create_children(Network & network,
             legal_sum += raw_netlist.policy[i];
         }
     }
-    nodelist.emplace_back(raw_netlist.policy_pass, FastBoard::PASS);
-    legal_sum += raw_netlist.policy_pass;
+
+    // Don't consider pass moves if they would be immediately losing and we "shouldn't"
+    // accept that; see issue #2273. --dumbpass disables this heuristic.
+    bool suppress_passes =
+        winrate_for_player >= 0.75         // we should be winning
+        && nodelist.size() >= 5            // there are surely good alternatives
+        && !cfg_dumbpass                   // allowed to be clever about passing
+        && state.final_score(to_move) < 0; // passing => immediate loss
```

Comment:
Suggest `<=0` as we want to support draws.

Reply:
Yeah, probably. I'll make that change. Some overengineering bit of my brain wants to consider whether we want a higher winrate threshold for drawn-if-passed positions than for lost-if-passed ones, but I think that's a wrinkle that can be added later in the unlikely event that it ever makes a difference.